File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.rb
# @param {String} s
# @return {String}
def reverse_parentheses(s)
  stack = []
  s.each_char do |char|
    if char == ')'
      temp = []
      while stack.last != '('
        temp << stack.pop
      end
      stack.pop #remove the '('
      stack.push(*temp)
    else
      stack.push(char)
    end
  end
  stack.join
end

File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseParentheses = function(s) {
  const stack = [];
  for (let char of s) {
    if (char === ')') {
      const temp = [];
      while (stack[stack.length - 1] !== '(') {
        temp.push(stack.pop());
      }
      stack.pop(); // remove the '('
      stack.push(...temp);
    } else {
      stack.push(char);
    }
  }
  return stack.join('');
};